python/pyarrow/tests/test_substrait.py

```diff
@@ -86,6 +86,22 @@ def test_run_serialized_query(tmpdir):
     assert table.select(["foo"]) == res_tb.select(["foo"])
 
 
+@pytest.mark.parametrize("query", (pa.py_buffer(b'buffer'), b"bytes", 1))
+def test_run_query_input_types(tmpdir, query):
+
+    # Passing unsupported type, like int, will not segfault.
+    if not any(isinstance(query, c) for c in (pa.Buffer, bytes)):
```

Comment:
FYI you can pass a tuple to check if the object matches any of the types:

```suggestion
    if not isinstance(query, (pa.Buffer, bytes)):
```

https://docs.python.org/3.6/library/functions.html#isinstance